packages/amplify-graphql-transformer-migrator/src/migrators/auth/defaultAuth.ts

```diff
@@ -8,13 +8,28 @@ const defaultAuthModeMap: Map<string, string> = new Map<string, string>([
   ['oidc', 'private']
 ]);
 
+const userPoolStrategies = new Set(['owner', 'groups']);
+
 function getDefaultAuthRule(authMode: any, rules: any) {
-  const parsedRules = rules.map((rule: any) => ({
-    provider: rule.fields.find((r: any) => r.name.value === 'provider')?.value?.value,
-    strategy: rule.fields.find((r: any) => r.name.value === 'allow')?.value?.value
-  }));
+  const parsedRules = rules.map((rule: any) => {
+    let provider = rule.fields.find((r: any) => r.name.value === 'provider')?.value?.value;
+    const strategy = rule.fields.find((r: any) => r.name.value === 'allow')?.value?.value;
+
+    if (userPoolStrategies.has(strategy) && !provider) provider = 'userPools';
+
+    return {
+      provider,
+      strategy,
+    };
+  });
 
   const foundRules = parsedRules.filter((r: any) => r.strategy === defaultAuthModeMap.get(authMode));
+  const userGroupRule = parsedRules.find((r: any) => userPoolStrategies.has(r.strategy));
+
+  if (userGroupRule) {
+    return userGroupRule;
+  }
```

Comment:
Can you move this block above the `foundRules` declaration. It seems like that would be wasted work if we return early here.